diff --git a/kirigami-plasmadesktop-integration/plasmadesktoptheme.cpp b/kirigami-plasmadesktop-integration/plasmadesktoptheme.cpp
index 51a3241f90e312b91c1abe2dcceb9abc2ffb974a..045264fdaf0496fb38bc0e5ef94cb317508279a9 100644
--- a/kirigami-plasmadesktop-integration/plasmadesktoptheme.cpp
+++ b/kirigami-plasmadesktop-integration/plasmadesktoptheme.cpp
@@ -50,14 +50,13 @@ public:
         connect(qGuiApp, &QGuiApplication::fontDatabaseChanged, this, &StyleSingleton::notifyWatchersConfigurationChange);
         qGuiApp->installEventFilter(this);
 
-        // Use NativeTextRendering as the default text rendering type when the scale factor is an integer.
-        // NativeTextRendering is still distorted sometimes with fractional scale factors,
-        // despite https://bugreports.qt.io/browse/QTBUG-67007 being closed.
-        qreal devicePixelRatio = qGuiApp->devicePixelRatio();
-        QQuickWindow::TextRenderType defaultTextRenderType =
-            int(devicePixelRatio) == devicePixelRatio ? QQuickWindow::NativeTextRendering : QQuickWindow::QtTextRendering;
-        QQuickWindow::setTextRenderType(defaultTextRenderType);
-
+        // NativeTextRendering is still distorted sometimes with fractional scale factors
+        // Given Qt disables all hinting with native rendering when any scaling is used anyway
+        // we can use Qt's rendering throughout
+        // QTBUG-126577
+        if (qApp->devicePixelRatio() != 1.0) {
+            QQuickWindow::setTextRenderType(QQuickWindow::NativeTextRendering);
+        }
         smallFont = loadSmallFont();
     }
 
