diff --git a/arch/x86/kernel/cpu/intel_epb.c b/arch/x86/kernel/cpu/intel_epb.c
index 30b1d63b9..5700a46ab 100644
--- a/arch/x86/kernel/cpu/intel_epb.c
+++ b/arch/x86/kernel/cpu/intel_epb.c
@@ -204,6 +204,8 @@ static int intel_epb_offline(unsigned int cpu)
 }
 
 static const struct x86_cpu_id intel_epb_normal[] = {
+	X86_MATCH_VFM(INTEL_ALDERLAKE,
+		      ENERGY_PERF_BIAS_NORMAL_POWERSAVE),
 	X86_MATCH_VFM(INTEL_ALDERLAKE_L,
 		      ENERGY_PERF_BIAS_NORMAL_POWERSAVE),
 	X86_MATCH_VFM(INTEL_ATOM_GRACEMONT,
